File: KnowledgeCenterClient/src/app/match/models/Collaborator.ts
import { Agency } from '../../shared/models/Agency';
import { ServiceLine } from '../../shared/models/ServiceLine';
import { CollaboratorSkill } from './CollaboratorSkill';

export interface Collaborator {
  id: number;
  ggid: number;
  firstname: string;
  lastname: string;
  email: string;
  fullname: string;
  agencyId: number;
  agency: Agency;
  serviceLineId: number;
  serviceLine: ServiceLine;
  collaboratorSkills: CollaboratorSkill[];
}

export interface CreateOrUpdateCollaborator {
  id: number;
  ggid: number;
  firstname: string;
  lastname: string;
  email: string;
  agencyId: number;
  serviceLineId?: number;
}

File: KnowledgeCenterServer/_Match/KnowledgeCenter.Match.Contracts/Collaborator.cs
using System.Collections.Generic;

namespace KnowledgeCenter.Match.Contracts
{
    public class Collaborator
    {
        public int Id { get; set; }

        public int? GGID { get; set; }

        public string Firstname { get; set; }

        public string Lastname { get; set; }

        public string Email { get; set; }

        public string Fullname
        {
            get
            {
                return $"{Firstname} {Lastname}";
            }
        }
        
        public int AgencyId { get; set; }

        public Agency Agency { get; set; }

        public int ServiceLineId { get; set; }
        
        public ServiceLine ServiceLine { get; set; }

        public List<CollaboratorSkill> CollaboratorSkills { get; set; }
    }
}